btree/btree.c

```diff
@@ -230,10 +230,115 @@
+ * step 6. MOVEUP: This state moves to the parent node and checks if there is an underflow in the parent node
+ *                 If there is an underflow move to RESOLVE, else move to CLEANUP.
+ * 
+ * 
  * Iteration (NEXT)
  * ................
- *
+ * @verbatim
+ * 
+ *                       INIT
+ *                         |
+ *                         v
+ *                       SETUP <----------------+----------------+
+ *                         |                    |                |
+ *                         v                    |                |
+ *                      LOCKALL<----------------+                |
+ *                         |                    |                |
+ *                         v                    |                |
+ *                       DOWN  <----------------+                |
+ *                         |                    |              UNLOCK
+ *                         v                    |                |
+ *                   +->NEXTDOWN---->LOCK---->CHECK              |
+ *                   |   |   |   ^              | <--------+     |
+ *                   +---+   v   |              |          |     |
+ *                          LOAD-+              +-------+  |     |
+ *                                              |       v  |     |
+ *                                             ACT----->NEXTKEY  |
+ *                                              |        |   ^   |
+ *                                              v        |   |   |
+ *                                           CLEANUP     v   |   |
+ *                                              |       NEXTNODE-+
+ *                                              v
+ *                                            DONE
+ * 
+ * @endverbatim
+ * Iteration function will look for a key and read keys one-by-one, if returned key is last key in the node , we may need to fetch next node,
+ * To fetch keys in next node: first release the LOCK, call Iteration function and pass key=last fetched key and next_sibling_flag=1
+ * If next_sibling_flag == 1, we will also load next sibling and , we need to fetch key next to given key
+ * As we are releasing lock for finding next node, updates such as(insertion of new keys, merging due to deletion can happen), 
+ * so in such cases, we will load both earlier node and node to the right of it
+ * 
+ * Phases Description: 
+ * step 1. NEXTDOWN: this state will load nodes searching by the given key if next_sibling_flag == 0, 
+ *                   but if next_sibling_flag == 1 then this state will load the leftmost child nodes during its downward traversal.
+ * step 2. LOAD: this function will get called only when next_sibling_flag == 1, 
+ *               and functionality of this function is to load next node so, it will search and LOAD next sibling 
+ * step 3. CHECK: check function will check the traversal path for node with key and traversal path for node next node if it is also loaded
```

Comment:
Please check this line. I don't understand this:
```
traversal path for node next node if it is also loaded
```

Reply:
it should be : traversal path for  next node if it is also loaded